components/transport-jms/src/test/java/org/wso2/transport/jms/test/client/polling/QueueClientPollingWithSelectorTestCase.java

```diff
@@ -0,0 +1,185 @@
+        connection = jmsServer.createConnection(null, null);
+        connection.start();
+
+        //publish 10 messages to the queue with correlationId1
+        publishToQueue(correlationId1, 10);
+        //publish 5 messages to the queue with correlationId2
+        publishToQueue(correlationId2, 5);
+
+        jmsClientConnectorQueue = new JMSConnectorFactoryImpl().createClientConnector(jmsClientParameters);
+    }
+
+    /**
+     * This test will publish 10 messages to the provided queue and consume them using JMS transport client polling
+     * interface and validate the number of produced messages.
+     *
+     * @throws JMSConnectorException  Error when polling messages through JMS transport connector.
+     * @throws JMSException           Error when running the test message consumer.
+     * @throws NoSuchFieldException   Error when accessing the private field.
+     * @throws IllegalAccessException Error when accessing the private field.
+     */
+    @Test(groups = "jmsPolling",
+          description = "Queue Polling with selectors test case")
+    public void queuePollingTestCase()
+            throws JMSConnectorException, NoSuchFieldException, IllegalAccessException, InterruptedException,
+            JMSException {
+
+        int receivedMsgCount1 = 0;
+        // this should receive 10 messages but loop will goon for two more iterations to see if any more messages comes
+        for (int i = 0; i < 12; i++) {
+            Message message = jmsClientConnectorQueue
+                    .poll(queueName, 500, "JMSCorrelationID = '" + correlationId1 + "'");
+            if (message != null) {
+                receivedMsgCount1++;
+            }
+        }
+
+        int receivedMsgCount2 = 0;
+        // this should receive 5 messages but loop will goon for two more iterations to see if any more messages comes
+        for (int i = 0; i < 7; i++) {
+            Message message = jmsClientConnectorQueue
+                    .poll(queueName, 500, "JMSCorrelationID = '" + correlationId2 + "'");
+            if (message != null) {
+                receivedMsgCount2++;
+            }
+        }
+
+        JMSTestUtils.closeResources((JMSClientConnectorImpl) jmsClientConnectorQueue);
+
+        Assert.assertEquals(receivedMsgCount1, 10,
+                "JMS Client selector polling receiver one expected message count " + 10 + " , received message count "
+                        + receivedMsgCount1);
+        Assert.assertEquals(receivedMsgCount2, 5,
+                "JMS Client selector polling receiver two expected message count " + 5 + " , received message count "
+                        + receivedMsgCount2);
+    }
+
+    @AfterClass(groups = "jmsPolling",
+                description = "Shutting down the server by closing the only connection")
+    public void cleanup() throws JMSException {
+        if (connection != null) {
+            connection.close();
```

Comment:
Don't we have to stop the server?

Reply:
This is an embedded broker, it will only be alive if there is any active connection. When we close the only active connection at the moment, broker dies.  
If you are referring to the Ballerina Server, at this test ballerina runs from a main program. therefore it will automatically terminate after executing the source

Reply:
+1